From 9130da8043b3747439ac06ff3c599a22f2376041 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 14:13:27 +0800
Subject: [PORT FROM MAIN] Extend MPEGExtractor to support "qt" type

BZ: 48862

Original patches:
R3: 30130 (BZ: 18544)
R2: 19843 (BZ: 10568)

1. Check if the major brand is 'qt ' in box 'ftyp' to support quicktime mp4 clips;
2. If there are some invaild data at the end of one box, skip them and keep parsing.

Change-Id: I9ca5f14d7fd559d0eaabcac1c4559dd6d830f0c3
Orig-Change-Id: I5f44aadfd97e40826da35f22d3d1a71dcd821288
Signed-off-by: Weian Chen <weian.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 796323d..e9a4581 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -743,7 +743,9 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
         }
     } else if (chunk_size < 8) {
         // The smallest valid chunk is 8 bytes long.
-        return ERROR_MALFORMED;
+        // Ignore the invalid data and continue
+        *offset += 4;
+        return OK;
     }
 
     char chunk[5];
@@ -3606,7 +3608,8 @@ static bool LegacySniffMPEG4(
         || !memcmp(header, "ftyp3ge6", 8) || !memcmp(header, "ftyp3gg6", 8)
         || !memcmp(header, "ftypisom", 8) || !memcmp(header, "ftypM4V ", 8)
         || !memcmp(header, "ftypM4A ", 8) || !memcmp(header, "ftypf4v ", 8)
-        || !memcmp(header, "ftypkddi", 8) || !memcmp(header, "ftypM4VP", 8)) {
+        || !memcmp(header, "ftypkddi", 8) || !memcmp(header, "ftypM4VP", 8)
+        || !memcmp(header, "ftypqt  ", 8)) {
         *mimeType = MEDIA_MIMETYPE_CONTAINER_MPEG4;
         *confidence = 0.4;
 
-- 
1.7.4.1

